Update docker-registry-sync

soggiest opened the pull request:
-Added an optional "project" parameter that will tag/push new images into the proper namespace
-Added code that will accept subrepos. Previously the code assumed images would be in <repository>/<image> format
  Example: <external repository>/jboss-amq-6/amq-openshift

Commit:
Update docker-registry-sync

-Added an optional "project" parameter that will tag/push new images into the proper namespace
-Added code that will accept subrepos. Previously the code assumed images would be in <repository>/<image> format
  Example: <external repository>/jboss-amq-6/amq-openshift

etsauer commented:
@soggiest can you pull in and fill out the PR template from the contribution guide? https://github.com/rhtconsulting/rhc-ose/blob/openshift-enterprise-3/CONTRIBUTING.md

soggiest commented:
#### What does this PR do?

-Added an optional "project" parameter that will tag/push new images into the proper namespace
-Added code that will accept subrepos. Previously the code assumed images would be in / format
Example: /jboss-amq-6/amq-openshift

#### How should this be manually tested?
- In an OSE environment create  a new project

```
oc project sync-test
```
- Create a file which contains an image to test syncing against

```
echo "jboss-amq-6/amq-openshift" > /tmp/sync-test
```
- Run the docker-registry script specifying a project

```
./docker-registry-sync --from=registry.access.redhat.com --local=<your local registry> --file=/tmp/sync-test --project=sync-test
```
- Verify that the new image exists in the OSE environment in the right project

```
oc get imagestreams -n sync-test
```

You should see output that looks similar to:

```
NAME            DOCKER REPO                               TAGS      UPDATED
amq-openshift   172.30.168.255:5000/sync-test/amq-openshift   latest    27 hours ago
```
- Run the same command without specifying a project

```
./docker-registry-sync --from=registry.access.redhat.com --local=<your local registry> --file=/tmp/sync-test
```

*Verify that the new image exists in the OSE environment in the openshift project

```
oc get imagestreams
```

You should see output that looks similar to:

```
NAME            DOCKER REPO                               TAGS      UPDATED
amq-openshift   172.30.168.255:5000/openshift/amq-openshift   latest    27 hours ago
```

#### Is there a relevant Issue open for this?

N/A

#### Who would you like to review this?

/cc @esauer

etsauer commented:
@soggiest Ah, interesting.

So this script was intended to work with a basic docker registry that was not the OpenShift registry. So I'm interested to see if the changes you've made still allow the script to work that way. I'm definitely open to having the ability to pre-load images in either an external or the openshift registry, as long as both work with this script.

Have you tested this against a standalone registry?

@oybed do you have any other thoughts on this?

soggiest commented:
I haven't tested it yet against a stand-alone registry. I'll test that shortly.

oybed commented:
No other thoughts. I agree that this change is a good change, and I like it!  I also agree with @etsauer that it needs to continue to work for stand-alone registries as that's how it is still being used with some customers.

etsauer commented:
@oybed @soggiest is this still a relevant PR? its been dormant for a while.
